src/config_store.rs

```diff
@@ -60,20 +59,27 @@ impl fmt::Display for Tick {
 
 /// The ConfigStore maintains the state of all check configurations and provides an efficient way
 /// to retrieve the checks that are scheduled for a given tick.
+///
+/// CheckConfigs are stored into TickBuckets based on their interval. Each bucket contains the
+/// checks that should be executed at that tick. Each configuration includes the partition it
+/// belongs to, the ConfigStore will group configurations by partition and supports dropping entire
+/// partitions of CheckConfigs.
 #[derive(Debug)]
 pub struct ConfigStore {
-    buckets: TickBuckets,
-    configs: CheckerConfigs,
+    /// A ConfigStore is derived from configurations loaded from Kafka, when the assigned
+    /// partitions for the uptime-checker are re-balanced, partitions will be added and removed.
+    /// Storing TickBuckets for each partition makes dropping an entire partition of configs during
+    /// removal simple.
+    partitioned_buckets: HashMap<i32, TickBuckets>,
+
+    /// Mapping of each subscription_id's config
+    configs: HashMap<Uuid, Arc<CheckConfig>>,
```

Comment:
I feel like the configs shouldn't be global, but should just be isolated inside each partition, that way they can't influence each other